Library/OcAppleKernelLib/CachelessContext.c

```diff
@@ -492,6 +578,148 @@ ScanDependencies (
   return EFI_SUCCESS;
 }
 
+STATIC
+VOID
+FreeGenericPatch (
+  PATCHER_GENERIC_PATCH       *Patch
+  )
+{
+  if (Patch->Comment != NULL) {
+    FreePool ((VOID*)Patch->Comment);
+  }
+  if (Patch->Base != NULL) {
+    FreePool ((VOID*)Patch->Base);
+  }
+  if (Patch->Find != NULL) {
+    FreePool ((VOID*)Patch->Find);
+  }
+  if (Patch->Replace != NULL) {
+    FreePool ((VOID*)Patch->Replace);
+  }
+  if (Patch->Mask != NULL) {
+    FreePool ((VOID*)Patch->Mask);
+  }
+  if (Patch->ReplaceMask != NULL) {
+    FreePool ((VOID*)Patch->ReplaceMask);
+  }
+
+  ZeroMem (Patch, sizeof (*Patch));
+}
+
+STATIC
+EFI_STATUS
+InternalAddKextPatch (
+  IN OUT CACHELESS_CONTEXT      *Context,
+  IN     CONST CHAR8            *BundleId,
+  IN     PATCHER_GENERIC_PATCH  *Patch,
+  IN     KERNEL_QUIRK_NAME      QuirkName
+  )
+{
+  PATCHED_KEXT          *PatchedKext;
+  KEXT_PATCH            *KextPatch;
+  KERNEL_QUIRK          *KernelQuirk;
+
+  if (Patch == NULL) {
+    KernelQuirk = &gKernelQuirks[QuirkName];
+    if (KernelQuirk->BundleId == NULL) {
```

Comment:
`ASSERT` maybe? Is this actually a runtime check here, i.e. can this be true under normal conditions?